samcli/commands/init/init_templates.py

```diff
@@ -171,3 +184,79 @@ def is_dynamic_schemas_template(self, package_type, app_template, runtime, base_
             if option.get("appTemplate") == app_template:
                 return option.get("isDynamicTemplate", False)
         return False
+
+    def get_preprocessed_manifest(self, filter_value=None):
+        """
+        This method get the manifest cloned from the git repo and preprocessed it.
+        The structure of the manifest is shown below:
+
+        {
+            "dotnetcore2.1": [
+                {
+                    "directory": "dotnetcore2.1/cookiecutter-aws-sam-hello-dotnet",
+                    "displayName": "Hello World Example",
+                    "dependencyManager": "cli-package",
+                    "appTemplate": "hello-world",
+                    "packageType": "Zip",
+                    "useCaseName": "Serverless API"
+                },
+            ]
+        }
+
+        Raises:
+            InvalidInitTemplateError: Exception raise when template is invalid or missing a parameter
+
+        Returns:
+            dict: This is preprocessed manifest with the use_case as key
+        """
+        if not self._git_repo.clone_attempted:
+            self.git_cloner()
+        manifest_path = os.path.join(self._git_repo.local_path, "manifest.json")
+        with open(str(manifest_path)) as fp:
+            body = fp.read()
+            manifest_body = json.loads(body)
+
+        preprocessed_manifest = {}
+
+        for template_runtime in manifest_body:
+            if filter_value and filter_value == template_runtime:
+                template_list = manifest_body[template_runtime]
+            elif filter_value and filter_value != template_runtime:
+                continue
+            else:
+                template_list = manifest_body[template_runtime]
+            for template in template_list:
+                validate_value_in_template("packageType", template)
+                package_type = template["packageType"]
+                runtime = get_runtime(package_type, template_runtime)
+
+                validate_value_in_template("useCaseName", template)
+                use_case = template["useCaseName"]
+                if use_case not in preprocessed_manifest:
+                    preprocessed_manifest[use_case] = {}
+
+                if runtime not in preprocessed_manifest[use_case]:
+                    preprocessed_manifest[use_case][runtime] = {}
+
+                if package_type not in preprocessed_manifest[use_case][runtime]:
+                    preprocessed_manifest[use_case][runtime][package_type] = []
+                preprocessed_manifest[use_case][runtime][package_type].append(template)
+        return preprocessed_manifest
+
+    def get_bundle_option(self, package_type, runtime, dependency_manager):
+        return self._init_options_from_bundle(package_type, runtime, dependency_manager)
+
+
+def validate_value_in_template(value, template):
```

Comment:
This is in an example of a validator function, collate your functions together like this one so that if conditions can be cleaned up.